pkgs/by-name/ar/argc/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  rustPlatform,
+  stdenv, # Determin platforms.
+  fetchFromGitHub,
+  nixpkgsFun, # Re-evaluate Nixpkgs.
+  argc, # Get native argc for shell completion generation during cross-compilation.
+  installShellFiles,
+}:
+
+let
+  canExecuteHost = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "argc";
+  version = "1.14.0";
+
+  src = fetchFromGitHub {
+    owner = "sigoden";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Li/K5/SLG6JuoRJDz2DQoj1Oi9LQgZWHNvtZ1HVbj88=";
+  };
+
+  cargoHash = "sha256-D1T9FWTvwKtAYoqFlR2OmLRLGWhPJ9D8J7lq/QKcBoM=";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional (!canExecuteHost) argc;
+
+  postInstall = ''
+    INSTALL_SHELL_FILE_PATH="${lib.optionalString canExecuteHost ''''${!outputBin}/bin''${PATH:+:}''}''${PATH-}"
+
+    PATH="$INSTALL_SHELL_FILE_PATH" installShellCompletion --cmd argc \
```

Comment:
We don't need to indirect the bin output here, as the package only has one output.

Instead of doing complex shell logic here, we can just turn argc into a variable, put it in a let in and set it to $out/bin/argc if we are not doing cross and otherwise to the binary for the executing platform.